app/src/main/java/ch/epfl/cs311/wanderwave/model/remote/ProfileConnection.kt

```diff
@@ -41,14 +47,26 @@ class ProfileConnection(private val database: FirebaseFirestore? = null) :
   }
 
   override fun itemToMap(profile: Profile): HashMap<String, Any> {
-    val profileMap: HashMap<String, Any> = profile.toMap()
-
+    val profileMap: HashMap<String, Any> = profile.toMap(db)
+    Log.d("ProfileConnection", "profileMap: $profileMap")
     return profileMap
   }
 
-  fun addProfilesIfNotExist(profiles: List<Profile>) {
+  override fun addItem(item: Profile) {
+    super.addItem(item)
+    trackConnection.addItemsIfNotExist(item.topSongs)
+    trackConnection.addItemsIfNotExist(item.chosenSongs)
+  }
+
+  override fun addItemWithId(item: Profile) {
+    super.addItemWithId(item)
+    trackConnection.addItemsIfNotExist(item.topSongs)
+    trackConnection.addItemsIfNotExist(item.chosenSongs)
+  }
+
+  fun addProfilesIfNotExist(profiles: List<Profile?>) {

```

Comment:
I'm not sure in what situation a user would even have permission to add multiple profiles to the database, what the usecase here? But not directly part of this PR.

Reply:
This was just to make sure that the trackAndProfile have some data but it's true that now that the profile can be empty in the track and profile assotiation, we can delete this line ! For another issue next week